File: javascript/insert_sort/insert_sort.js

function insertSort(ary) {
  let len = ary.length;

  for (let i = 1; i < len; i++) {
      let j = i;
      for (; j > 0 && ary[j] < ary[j-1]; j--) {
        let temp = ary[j];
        ary[j] = ary[j-1];
        ary[j-1] = temp;
      }
  }
}

let minArrayLen = 2;
let maxArrayLen = 10;
let valRange = 1000;
for (let i = minArrayLen; i <= maxArrayLen; i++) {
  let ary = new Array(i)
    .fill(true)
    .map(e => parseInt(Math.random() * valRange))
  console.log(ary);
  insertSort(ary);
  console.log(ary);
  console.log();
}

let ary = [ 10, 9, 8, 7, 6, 5, 4, 3, 2, 1]
console.log(ary)
insertSort(ary)
console.log(ary)

File: python/insert_sort/insert_sort.py
import random

aryLen = 10
ary = range(1,aryLen+1)

def knuthShuffle(ary):
  aryLen = len(ary)
  for i in range(1, aryLen+1):
    j = aryLen - i
    if j > 0:
      swapI = random.randint(0, j-1)
    else:
      swapI = 0
    temp = ary[swapI]
    ary[swapI] = ary[j]
    ary[j] = temp

def insertSort(ary):
  aryLen = len(ary)
  for i in range(1, aryLen):
    j = i
    while j > 0 and ary[j] < ary[j-1]:
      temp = ary[j]
      ary[j] = ary[j-1]
      ary[j-1] = temp
      j -= 1

def insertSortDesc(ary):
  aryLen = len(ary)
  for i in range(1, aryLen):
    j = i
    while j > 0 and ary[j] > ary[j-1]:
      temp = ary[j]
      ary[j] = ary[j-1]
      ary[j-1] = temp
      j -= 1



print 'created', ary
knuthShuffle(ary)
print 'shuffled', ary
insertSort(ary)
print 'insert sort asc', ary
knuthShuffle(ary)
print 'shuffled', ary
insertSortDesc(ary)
print 'insert sort desc', ary